### ISSUE TITLE
Reflog always recreated when running `cvmfs_server check -r -i  FQRN`

### ISSUE THREAD

**Issue by olifre** [cvmfs/cvmfs#3123]
I think this issue is present since CVMFS 2.8.0, but I got aware of it just now:

When running `cvmfs_server check -r -i  FQRN` against a repository which is in "good" state, it seems the reflog is always reconstructed. I presume it is related to:
https://github.com/cvmfs/cvmfs/commit/8f2d9167fc74ee246b270ddda68cae5012f403cc

Checking the logic of the code, it seems that reflog reconstruction is always started even if `.cvmfsreflog` and `reflog.chksum` both exist and are in sync. 

Logs of such a case:
```
Checking storage integrity of container.physik.uni-bonn.de ... (may take a while)
Verifying integrity of container.physik.uni-bonn.de...
Info: found http://HOST/cvmfs/container.physik.uni-bonn.de/.cvmfsreflog with content hash bf796fc4d6cda7a250d5d5edaff7ef47e4030ac5-shake128
Warning: a reflog hash is registered in the manifest, re-creating missing reflog
```
Syslog contains:
```
Dec  4 09:10:05 cvmfs-stratum-one-2 cvmfs_server: (container.physik.uni-bonn.de) reference log reconstruction started
Dec  4 09:11:45 cvmfs-stratum-one-2 cvmfs_server: (container.physik.uni-bonn.de) successfully reconstructed reference log
```

Checking `Y` key:
```
[root@cvmfs-stratum-one-2 ~]# cvmfs_swissknife info -r http://localhost/cvmfs/container.physik.uni-bonn.de/ -R | grep '^Y'
Y31eb22a2e3d32b1ea31761f143826a4b9d85c163-shake128
```
Checking hashes:
```
$ curl -sS --fail --connect-timeout 10 --max-time 300 http://localhost/cvmfs/container.physik.uni-bonn.de/.cvmfsreflog | cvmfs_publish hash -a shake128
b2a02cba3c4f1907ef5479a0c94858b58aae5644-shake128
$ cat /var/spool/cvmfs/container.physik.uni-bonn.de/reflog.chksum
b2a02cba3c4f1907ef5479a0c94858b58aae5644-shake128
```

So hashes are in sync, the `Y` key exists, but the reflog is still recreated.

**Comment by olifre** [comment]
After some thinking, I believe the condition `[ $has_reflog -eq 0 ]` is missing from the last `if` condition, i.e.:
```bash
if get_repo_info -R | grep -q ^Y; then
```
should be:
```bash
if [ $has_reflog -eq 0 ] && get_repo_info -R | grep -q ^Y; then
```
instead, if my understanding of the code is correct.

**Comment by jblomer** [comment]
Many thanks for the detailed report! I think you're right about the fix. PR is up. If it passes the tests, I'll backport it to the 2.10 branch.

**Comment by olifre** [comment]
Many thanks for the fix and taking it up for backporting to 2.10, glad my understanding of the code was correct :smile:.
